### DISCUSSION TITLE
add_links("cuda") if you want to link cuda driver library

### DISCUSSION THREAD

**Discussion by Extra-Creativity** [xmake-io/xmake#4623] [General]
`add_package("cuda")` will only link against cuda runtime library; don't forget to `add_links("cuda")` if you want to use cuda driver API.

This isn't a flaw of xmake; cmake doesn't link against libcuda.so/cuda.dll either.

**Comment by xq114** [comment]
Both `add_requires("cuda", {configs={utils={"cuda"}}})` and manual `add_links` would work. Since there are too many cuda libraries, it should require a setting anyway
